### PR TITLE
Fix game search errors and hiding carousel errors

### PR DESCRIPTION
**Pull request by LINK1733**
Updated node & npm on my computer since it was rebuilding the package lock on the hosted version since I was on an older version.

If you had special characters such as a colon in the game search it would error out so I changed the search input regex to replace them with an underscore and also trimmed the start and end of underscores that may be there after the first replacement.

Added a check for the hl2b times so it won't show the arrows if there are no game times at all or before the game time carousel is rendered.

Reduced padding on all lists and changed the game to do lists style to match the game lists.

There were also a couple things I was pretty sure I changed for webstorm warnings that seemed to change back for some reason so they were resolved again.

Picture of the new padding is below

image

### CODE REVIEW

File: src/server/controllers/gameList.js [hunk 1]
```diff
@@ -27,18 +27,22 @@ module.exports.getGameList = catchAsync(async (req, res) => {
 module.exports.searchGame = catchAsync(async (req, res) => {
 	const gameListGames = req.body.gameListGames;
 
+	let searchInput = req.body.searchInput.replace(/[^0-9a-zA-Z]+/g, '_');
+
+	searchInput = searchInput.replace(/^_|_$/g, '');
```
Comment (sheodox) [inline src/server/controllers/gameList.js:32]: FYI you can just chain replaces like const searchInput = str.replace().replace()

I generally put each replace on a new line too.

File: src/server/controllers/users.js [hunk 2]
```diff
@@ -76,7 +76,7 @@ module.exports.updatePassword = async (req, res) => {
 				(await bcrypt.compare(oldPassword, currentUser.hash));
 
 		if (passwordCorrect) {
-			newHash = await hashPassword(newPassword);
+			let newHash = await hashPassword(newPassword);
```
Comment (sheodox) [inline src/server/controllers/users.js:79]: Should probably be a const right? Fwiw if you use a let when you don't ever reassign it eslint will complain, another reason to use eslint someday!

**Review by sheodox — APPROVED** [review]

File: src/server/controllers/users.js (hunk 2 above)
Comment (LINK1733) [reply src/server/controllers/users.js:79]: Good point, I hadn't thought about that. I changed this now.

File: src/server/controllers/gameList.js (hunk 1 above)
Comment (LINK1733) [reply src/server/controllers/gameList.js:32]: Makes sense. I think I'm with you where I'll leave it like this so it's a bit easier to read.

**Pull request merged**